### ISSUE TITLE
[REQUEST] Add support for Applied Energistics 2 - rv3 beta to v10.

### ISSUE THREAD

**Issue by Haggle1996** [ReikaKalseki/Reika_Mods_Issues#481]
Now that AE2 rv3 has entered beta after a long, stable series of alpha releases, I'd like to request you update compatibility during the v10 lifetime.

**Comment by ReikaKalseki** [comment]
What changed?

**Comment by Haggle1996** [comment]
I only see one real API change according to: http://ae2.ae-mod.info/Downloads/ and https://github.com/AppliedEnergistics/Applied-Energistics-2/commits/master/src/api/java/appeng/api otherwise it's just some enumeration expansions for other mods.
- No longer exposes internal exceptions through the API

Seems like that would most likely be swapping some typed exceptions, if you even handled them.

**Comment by ReikaKalseki** [comment]
That is not a substantial change, but I am leery of updating yet. As it is look at the s***show my thread is for people complaining that I broke their word (still using Forestry 3.0).

**Comment by Haggle1996** [comment]
IMHO, the silent majority wants you to update, we're just patient and smart enough not to pester you. :)

I have a whole slew of mod updates pending to fix AE2/AE2 Addon problems, but I can't push those fixes until your mods update (and yours is the only one that hasn't updated to rv3). 

Of the comments I've read complaining about your update to Forestry 4, they seem to be universally coming from players trying to shove your mods into a pack that wasn't designed to include RotaryCraft, et al., versus actively curated, popular RotaryCraft/ChromatiCraft packs from people like myself and @reteo. I don't know Survival Industry's stats, but my pack averages around 1000 new installs a day. I get really exhausted answering the same 'AE2 crashes when I do x; plz fix kthx' several times a day, particularly when there's an update available.

TL;DR; generally better to move forward than to hold back, particularly with modding.

**Comment by reteo** [comment]
I wish I had something to offer to this conversation, but 1: I don't know how many people play Survival Industry, since FTB lacks the tracking features of ATLauncher, and 2: Nobody whose played the modpack (including myself) has reached the point to build a ME network for the issues to surface.

**Comment by ghost** [comment]
> I don't know Survival Industry's stats, but my pack averages around 1000 new installs a day.

I don't know Survival Industry's stats, either; The Feed the Beast launcher doesn't seem to have a tracker (or else I don't know how to access it).  But so far, nobody playing it (including myself) has even attempted AE yet. :-/  And even so, I've been keeping it at the stable version to avoid having to put fires out.

**Comment by Sima214** [comment]
Also just a note for those still having **java.lang.NoClassDefFoundError: Could not initialize class Reika.ChromatiCraft.Registry.ChromaResearchManager$ResearchLevel** type crashes with rv3 of AE, you can disable some stuff in the config **AppliedEnergistics2/CustomRecipes.cfg** to avoid it.
The game then just loads up normally, however I don't know if any intergration is broken.

**Comment by ReikaKalseki** [comment]
What do you need to disable to fix it?

**Comment by Sima214** [comment]
(All these configs can be found under the general category in [minecraft_directory]/config/AppliedEnergistics2/CustomRecipes.cfg)
First of I've disabled both "enabled" and "exportItemNames",as those are the values I use.
After some testing it seems like the minimal you have to disable is "exportItemNames".

**Comment by ReikaKalseki** [comment]
Very, very strange...

**Comment by Keyarose** [comment]
Just some additional information about this issue, if you try to start minecraft again after the crash Sima214 mentioned it will work without making the changes in the AE2 config.  However the next time you add or remove a mod the crash could happen again, but the second try will work.  At least this is how it's been for me.

**Comment by ReikaKalseki** [comment]
Does AE have any sort of "on first installation" behavior that resets on pack change?

**Comment by TomeWyrm** [comment]
The facade compatability I would assume. Probably other things as well, but I cannot think of them.

**Comment by ReikaKalseki** [comment]
Did any of it change in rv3?

**Comment by TomeWyrm** [comment]
Is the custom recipe config thing new to rv3?

**Comment by Ianaver** [comment]
I was able to reload the game directly after the crash and it works just fine.... it still sucks since I do find that when I actually remove or add a mod with chromaticraft, I get that crash and have to start the game twice for it to reload properly

**Comment by ReikaKalseki** [comment]
I have no idea.

**Comment by Haggle1996** [comment]
I believe Custom Recipes were added in rv2, but were disabled by default. Turning off Custom Recipes allows RV3 to work fine with Reika's mods, and I have been using it for some time in both of my packs. There are alternative ways of accomplishing what Custom Recipes did, so there's no complete loss of functionality.

As the original reporter, I would vote to close this issue, but Reika may have it open for other reasons.

**Comment by ReikaKalseki** [comment]
"Export item names" can also be disabled to fix it.
